lexer/impl/lexer.cpp

```diff
@@ -1,6 +1,309 @@
+        }
+        else
+        {
+            // check not tab sequence
+            if (int(word[0]) == 0 || isspace(word[0]))
+            {
+                return 0; // take next word
+            }
+            // read char-by-char:
+            int index = 0;
+            std::string sign;
+            TokenType val;
+            while (index < int(word.length()))
+            {
+                char c = word[index++];
+                switch (c)
+                {
+                // single-char tokens
+                case '+':
+                case '-':
+                case '%':
+                case '*':
+                case '(':
+                case ')':
+                case '[':
+                case ']':
+                case '=':
+                case ';':
+                case ',':
+                    sign = toStr(c);
+                    val = trie.Find(sign).value_or(invalidToken);
+                    result.push_back(makeToken(val, lineNum, sign));
+                    break;
+                case '.':
+                    checkTwoCharToken(&index, word, "..", TokenType::Dot, TokenType::TwoDots);
+                    break;
+                case '<':
+                    checkTwoCharToken(&index, word, "<=", TokenType::SmallerComp, TokenType::SeqComp);
+                    break;
+                case '>':
+                    checkTwoCharToken(&index, word, ">=", TokenType::BiggerComp, TokenType::BeqComp);
+                    break;
+                case '/':
+                    checkTwoCharToken(&index, word, "/=", TokenType::DivOp, TokenType::NeqComp);
+                    break;
+                case ':':
+                    checkTwoCharToken(&index, word, ":=", TokenType::Column, TokenType::Assignment);
+                    break;
+
+                default:
+
+                    if (isdigit(c))
+                    { // if number constant
+                        index--;
+                        std::string num = readWhileNum(&index, word);
+                        TokenType type = TokenType::IntegerLiteral;
+                        if (index < int(word.length()) && word[index] == '.')
+                        { // if real
+                            int oldIndex = index;
+                            index++;
+                            std::string floatPart = readWhileNum(&index, word);
+                            if (floatPart.length() < 1)
+                            { //eg ..
+                                index = oldIndex;
+                                result.push_back(makeToken(type, lineNum, num));
+                                break;
+                            }
+                            num.append(".");
+                            num.append(floatPart);
+                            type = TokenType::RealLiteral;
+                        }
+                        result.push_back(makeToken(type, lineNum, num));
+                    }
+                    else if (iswalpha(c))
```

Comment:
DO NOT use wide-character versions of character-specific functions, streams, etc. from the std library. They do not behave in a way you would expect, and not portable in a way you would expect.
```suggestion
else if (std::isalpha(c)) {
```